pkgs/by-name/zl/zluda/package.nix

```diff
@@ -40,16 +40,28 @@ rustPlatform.buildRustPackage rec {
   preConfigure = ''
     # Comment out zluda_blaslt in Cargo.toml
     sed -i '/zluda_blaslt/d' Cargo.toml
+    # TODO: investigate test failure (the test seems to require build time env vars that aren't set on linux?)
+    rm zluda_inject/tests/inject.rs
   '';
 
   buildPhase = ''
     runHook preBuild
     cargo xtask --release
     runHook postBuild
   '';
+  # xtask doesn't support passing --target, but nix hooks expect the folder structure from when it's set
+  env.CARGO_BUILD_TARGET = stdenv.hostPlatform.rust.cargoShortTarget;
 
   cargoHash = "sha256-gZdLThmaeWVJXoeG7fuusfacgH2RNTHrqm8W0kqkqOY=";
 
+  # vergen panics if the .git directory isn't present
+  # Disable vergen and manually set env
+  postPatch = ''
+    substituteInPlace zluda/build.rs \
+      --replace-fail 'vergen(Config::default())' 'Some(())'
+    '';
+  env.VERGEN_GIT_SHA = "1b9ba2b2333746c5e2b05a2bf24fa6ec3828dcdf-nix";
```

Comment:
Where is this hash from

Reply:
Can't we just make them all zeros

Reply:
@jcaesar ^

Reply:
It's from https://github.com/vosen/ZLUDA/releases/tag/v3. My idea was to set it to something that upstream would recognize as "version 3 but packaged by nix" if somebody pasted some logs into a GH issue on their repo.  
I think the hash is used to invalidate cache when the installed version of zluda changes, so all zeros would be a bit bad. But what I initially suggested also isn't good, since changing it might be forgotten.

How about changing it to `v${version}-nix` or so?

Reply:
@SomeoneSerge opinions about the above? Is it risky cache-wise to use all zeroes or would it be ok idea to use non-SHA custom version string?

Reply:
Sorry for dropping out of the loop. I don't think this matters a whole lot, as you said it's probably only used for logging, so at most we risk confusion. With the value manually specified in `env.VERGEN_GIT_SHA` it'll likely get out of sync with `src` at some point (we'll just forget to update it) which will create more confusion than an explicit fake all-zeros sha1.

Reply:
`src.rev` sounds good.

Reply:
Changed to use the `src.rev` in the latest commit